charts/consul/templates/datadog-agent-rolebinding.yaml

```diff
@@ -0,0 +1,22 @@
+{{- if .Values.global.metrics.datadogIntegration.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
```

Comment:
**suggestion:** I don't think the metadata is 100% correct. It should be similar to the other rolebindings and shouldn't the component be `server` instead of `agent`?

I'd use `datadog-metrics` throughout and wouldn't use agent as it is such an overloaded term and we do not have agents anymore in consul-k8s.

```
metadata:
  name: {{ template "consul.fullname" . }}-datadog-metrics
  namespace: {{ .Release.Namespace }}
  labels:
    app: {{ template "consul.name" . }}
    chart: {{ template "consul.chart" . }}
    heritage: {{ .Release.Service }}
    release: {{ .Release.Name }}
    component: server
```

Reply:
> shouldn't the component be server instead of agent?

This label is referencing the datadog-agent, and therefore the role is pertinent to the dd agent in this case.

Updating the the other labels though